Chess/src/Bishop.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Bishop.h"
+
+// It moves and captures along diagonals 
+vector<pair<int, int>> Bishop::getBasicMoves()
+{
+	return Bishop::getValidMoves();
+}
+
+std::vector<std::pair<int, int>> Bishop::getValidMoves()
+{
+	vector<pair<int, int>> validMoves;
```

Comment:
Why search for all the standard moves if it is not required? It is not better to check whether the specific slot is suitable for the next step